docs/monadtransformers/index.md

```diff
@@ -0,0 +1,213 @@
+```scala mdoc:silent
+abstract class BaseException(message: String) extends Exception(message)
+
+case class UserNotFoundException(message: String) extends BaseException(message)
+
+def getUserById(userId: Int): Try[Option[User]] = { /* ... */ ??? }
+
+def ensureUserExists(userId: Int): EitherT[Try, BaseException, User] = {
+  OptionT(getUserById(userId))
+    .toRight(left = UserNotFoundException(s"user not found, userId=$userId"))
+}
+```
+
+`toRight` is pretty analogous to the method `Either.fromOption` mentioned before: just as `fromOption` built an `Either` from an `Option`, `toRight` creates an `EitherT` from an `OptionT`.
+If the original `OptionT` stores `Some` value, it will be wrapped into `Right`; otherwise the value provided as the `left` parameter will be wrapped into a `Left`.
+To provide the `left` value within the monad, there is corresponding `toRightF` method.
+
+`toLeft` is `toRight`'s counterpart which wraps the `Some` value into `Left` and transforms `None` into `Right` enclosing the provided `right` value.
+This is less commonly used in practice, but can serve e.g. for enforcing uniqueness checks in code.
+We return `Left` if the value has been found, and `Right` if it doesn't yet exist in the system.
+
+The methods available in `EitherT` are pretty similar to those we've seen in `OptionT`, but there are some notable differences.
+
+You might get into some confusion at first when it comes to e.g. `map`.
+In the case of `OptionT`, it was pretty obvious what should be done: `map` should go over the `Option` enclosed within `Try`, and then map the enclosed `Option` itself.
+This is slightly less obvious in case of `EitherT`: should it map over both `Left` and `Right` values, or only the `Right` value?
+
+The answer is that `EitherT` is _right-biased_, therefore plain `map` actually deals with the `Right` value.
+
+Having said that, let's take a quick look at the right-biased methods that `EitherT` offers:
+
+| Method | Takes | Returns |
+| :---: | --- | --- |
+| `map[D]` | `B => D` | `EitherT[F, A, D]` |
+| `subflatMap[D]` | `B => Either[A, D]` | `EitherT[F, A, D]` |
+| `semiflatMap[D]` | `B => F[D]` | `EitherT[F, A, D]` |
+| `flatMapF[D]` | `B => F[Either[A, D]]` | `EitherT[F, A, D]` |
+| `flatMap[D]` | `B => EitherT[F, A, D]` | `EitherT[F, A, D]` |
+
+As a side note, there are also certain methods in `EitherT` (that you're likely to need at some point) which map over the `Left` value, like `leftMap`, or over both `Left` and `Right` values, like `fold` or `bimap`.
+
+`EitherT` is very useful for fail-fast chained verifications:
+
+```scala mdoc:silent
+
+case class Item(state: String)
+class ItemOrder  { /* ... */ }
+
+case class ItemNotFoundException(message: String) extends BaseException(message)
+case class InvalidItemStateException(message: String) extends BaseException(message)
+
+def getItemById(itemId: Int): Try[Option[Item]] = { /* .. */ ??? }
+
+def ensureItemExists(itemId: Int): EitherT[Try, BaseException, Item] = {
```

Comment:
Should we update this example to also not return the transformer on these methods?

Also, do we even need this example here on the index page? Perhaps this could be moved to the EitherT page?

Reply:
Agreed. I copied it from the original, but it isn't adding much to this page.